### ISSUE TITLE
WARNING: CCXT: unable to create order poloniex / STEEM/ETH: poloniex POST https://poloniex.com/tradingApi HTTPSConnectionPool(host='poloniex.com', port=443): Read timed out. (read timeout=10)

### ISSUE THREAD

**Issue by trungtv** [enigmampc/catalyst#315]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System: (Windows Version or `$ uname --all`)
Darwin -Pro.local 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64

* Python Version: `$ python --version`
Python 3.6.3 :: Anaconda custom (64-bit)
* Python Bitness: `$ python -c 'import math, sys;print(int(math.log(sys.maxsize + 1, 2) + 1))'`
64
* How did you install Catalyst: (`pip`, `conda`, or `other (please explain)`)
pip
* Python packages: `$ pip freeze` or `$ conda list`

Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue
My algo created an order to sell some on poloniex. Due to read timeout, the order get to be executed twice. Which is the best way to configure ccxt timeout in Catalyst? 
```
WARNING: CCXT: unable to create order poloniex / STEEM/ETH: poloniex POST https://poloniex.com/tradingApi HTTPSConnectionPool(host='poloniex.com', port=443): Read timed out. (read timeout=10)
[2018-04-27 02:54:38.126355] WARNING: exchange_blotter: Ordering again.
```
Sincerely,
`$ whoami`

**Comment by lenak25** [comment]
Hi @trungtv ,

Thanks for catching this, we are working to solve this to handle the timeout exception properly.

Lena

**Comment by AvishaiW** [comment]
Hi, I added a code that is suppose to fix such exceptions and which should avoid duplicate orders to the develop branch. tests are still needed in order to make sure it works properly, once they are done, we will release this.

**Comment by lenak25** [comment]
Fixed at release 0.5.10.
